torch/backends/_nnapi/serializer.py

```diff
@@ -1549,19 +1549,17 @@ def add_adaptive_avg_pool2d(self, node):
         self.add_operation(NNAPI_OperationCode.AVERAGE_POOL_2D, inputs, outputs)
 
     def add_upsample_nearest2d(self, node):
-        assert node.inputsSize() == 3 or node.inputsSize() == 4
+        assert node.inputsSize() == 4
         assert node.outputsSize() == 1
-        if node.inputsSize() == 3:
-            image, size_jit, scale_jit = node.inputs()
-        else:
-            image, size_jit, scale_h_jit, scale_w_jit = node.inputs()
-        size_ctype, size_arg = self.get_constant_value(size_jit)
 
-        if node.inputsSize() == 3:
-            scale_ctype, scale_arg = self.get_constant_value(scale_jit)
+        image, size_jit, arg2_jit, arg3_jit = node.inputs()
+
+        arg3_ctype, arg3_arg = self.get_constant_value(arg3_jit)
+        if arg3_ctype.kind() == "BoolType":
+            scale_ctype, scale_arg = self.get_constant_value(arg2_jit)
         else:
-            scale_h_ctype, scale_h_arg = self.get_constant_value(scale_h_jit)
-            scale_w_ctype, scale_w_arg = self.get_constant_value(scale_w_jit)
+            scale_h_ctype, scale_h_arg = self.get_constant_value(arg2_jit)
+            scale_w_ctype, scale_w_arg = self.get_constant_value(arg3_jit)
```

Comment:
Just to check, does the new rounding logic need to be added [here in this function](https://github.com/pytorch/pytorch/blob/7c3b66d277ff58250d6fdb56ca4480df4cb6a30a/torch/backends/_nnapi/serializer.py#L1606-L1607) as well? I think python `int()` will round towards zero